### ISSUE TITLE
Postmark mail template not working

### ISSUE THREAD

**Issue by aldarund** [anymail/django-anymail#121]
* Anymail version 4.1
* ESP (Mailgun, SendGrid, etc.) Postmark
* Versions of Django, requests, python 2.1
* Exact error message and/or stack trace
```
Postmark API response 422: Unprocessable Entity
{
  "ErrorCode": 1123,
  "Message": "Neither 'TextBody', 'HtmlBody', or 'Subject' may be specified when using a template for the email."
}
```
```
 msg = AnymailMessage(subject=None, body=None, from_email=settings.DEFAULT_FROM_EMAIL,
                             to=to)
 msg.merge_global_data = context
 msg.template_id = 123456
 msg.send()
```
The problem is that django set body to '' even if None is passed
https://github.com/django/django/blob/b9cf764be62e77b4777b3a75ec256f6209a57671/django/core/mail/message.py#L234
But postmark dont like even empty string and dont send such messages.
So the solution would be to set body to None in Postmark backend if template_id is set.

**Comment by medmunds** [comment]
Thanks, good catch. I'm updating Anymail as you suggest. In the meantime, a workaround is to set `body` to `None` after creating the message, to override Django's default empty string:

```python
msg = AnymailMessage(subject=None, to=to, merge_global_data=context, template_id=123456)
msg.body = None
msg.send()
```

(From my notes, it seems like Postmark maybe used to ignore an empty TextBody with a template send, so this error message might be a recent change. And from that error message, it seems like they also eliminated the ability to override the template subject in the API call. But it's possible my notes weren't accurate.)

**Comment by medmunds** [comment]
Fix released in Anymail v4.2.
